### PR TITLE
Ness Kit touchups + Lucas Tweaks

### PR DESCRIPTION
**Pull request by O1yyy**
assets: 

[pk_assets.zip](https://github.com/HDR-Development/HewDraw-Remix/files/12286956/pk_assets.zip)

## also thank you jaccob101 for the Lucas Magnet Hitlag tweaks ~~and Ness DThrow FAF edits,~~ an old pr is linked here: https://github.com/HDR-Development/HewDraw-Remix/pull/1872

high level interpretation of my changes: 

# ness

### pkthunder
- pkt2 blue screen removed
- pkt1's hover is much better, negligible fall acceleration now
- turning angle on pkt1 ball buffed, should make chosing angles feel a little better
- you can drift much sooner after pkt2 ends
- fixed wall collision, no longer dies on walls
- now grabs ledge behind him while at peak pkt2 (nutty underside angles are now possible)
- pkthunder on both kids now can edge slipoff on startup and edge cancels at the peak
- decreased initial speed: 3.7 -> 3.5 to account for wall jump

### added a wall jump
- decreased vanilla jump height
- passive_wall_jump_y_speed: 2.5 ->1.9

### ~~pkfire~~
~~- pkfire no longer recoils~~
- nvm job did it

### magnet
- fixed the jc frame, JC on F8 (hitbox still on F7)
- hitlag mult 1.0x -> 0.5x (JC after 3 frames of hitlag: F10)
- removed set kb & 361 angle, switched to kbg/bkb, lessened bkb and gave it kbg
  - angle 361 -> 55
  - FKB: 70 -> 0
  - KBG: 3 -> 135
  - BKB: 20 -> 10
- cleaned up effects
- added a helper hitbox to help send opponents forward more reliably

image

### nair
- pulled visual into the torso to better match the hitboxes
- Early hit activity F5-F12 -> F5->F9
- Late hit activity F13-F16 -> F9 -> F16
- Late hit KBG 100 -> 90
- Late hit BKB 15 -> 25

### Dair:
- Late hit activity F24-F30 -> F24-F29

### fair
- multihits
  - angle 365 -> 361
  - tweaked kb to compensate for pm esque behaviour
  - cleaned up acmd so the gaps between hits are consistent
  - added a helper hitbox covering the lenght of his body
    - same stats as main psi hitbox
  
https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/09d49494-2cf3-49fb-9c36-98f2a4cbe7d8

### usmash
- fixed double hit usmash
- fixed no hitbox when uncharged 

### dsmash
- cleaned up hitboxes to fix blindspots, comparable to pm now

### dthrow
~~- FAF 50 -> 45~~
- kbg: 85 -> 65
- bkb: 70 -> 80

### jab 1
- Angle changed. 361 -> 70 (jablock still 361)
- Damage 2.0 -> 3.0
- KBG 15/25 -> 25
- Hitlag mul 1.7 -> 1.0
- Adjusted G/A Priorities for jablock/normal hitbox.

### jab 2
- Angle changes 361 -> 55 (jablock still 361)
- Damage 2.0 -> 3.0
- KBG 20/25 -> 25.
- Jablock hitlag mul 1.2 -> 1.0
- Adjusted G/A Priorities for jablock/normal hitbox.

### dash attack
- Single hit 1: damage 7.0 -> 8.0

- Hit 1 ~
  - damage 4.0/3.0 -> 4.0
  - Angle changes 0/45 -> 0/361
  - SDI muls 1.0 -> 0.5

- Hit 2 ~
  - SDI Mul 1.0 -> 0.5

- Hit 3 
  - Hitlag mul 1.5 -> 1.3

### dtilt
- Dtilt adjusted to P+
  - angle 361 -> 83
  - inner hitbox damage 4.5% -> 3.0%, does same damage allover
  - KBG: 3 -> 135
  - BKB: 20 -> 10
  - hitlag mult: 1.0 -> 0.85
  - sdi mult: 1.0 -> 0.5

# lucas
### zair
- active on frames f9-22 -->f7-10
- damage: 4% --> 8%
- angle: 70
- kbg: 70
- bkb: 30
- hitlag mult: 1.3x
- ledge box check: frame 7 --> 6

https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/6d1e3828-26cc-41cc-93fb-dcfde34bfb68

### up b
- multihits
  - now interpolate properly
- id 0: (front head hitbox)
  - damage: 1.0% -> 2.0%
  - angle: 365 -> 366
  - kbg: 100 -> 130
  - fkb: 180 -> 50
  - sdi mult: 1.5x -> 0,5x
- id 1: (back end hitbox)
  - damage: 2.0% -> 1.0% (encourage sdi attempts)
  - same stats as id 1
- finisher
  - matched hitbox layout to multihits
  - size of id 0: 12.5 -> 8.5

image

* NOT TO SCALE, but should give a rough idea
* these changes end up not doing much to sdi'ing the move (basically its still easy to get out with sdi and di in), since lucas is moving forward relatively quickly, they only aim to make the move feel more consistent

### charge burst freeze rest
- no longer a rest, you need to confirm off of something to kill with it
  - kbg 115 -> 60
  - bkb 50 -> 60

### pkfreeze side b
- removed ability to drift and fast fall
- to reliably rapid fire/use it, you need to use your dj to djc and land at a similar timing
  - this change is conpensated by the fact that djc side b is much more accessible in this engine with both procons and button remaps available

https://github.com/HDR-Development/HewDraw-Remix/assets/19886440/42851d13-6aaf-43b2-a6e0-754206c0624e

### dtilt
- pushed active frames from f3-4 -> f5-6
- added another sourspot hitbox to cover the gap between the two hitbox a little better
  - concequently makes sweetspot harder to get
  
2023090102133600-0E7DF678130F4F0FA2C88AE72B47AFDF

### nair
- cleaned up the sound script

### magnet
- (from jaccob101) Matches Magnet hitlag to p+ for input parity. Now should be 1 hitlag on the hold hitbox, and 4 frames on the release hitbox. Previously the release had the 2 frames of hitlag and the hold hitbox had 5 frames. Counting the frame it hits.
  - JC on hitlag: F11
### double jump
- shifted movement arcs back 2 frames inorder to facilitate perfect djc's and general djc feel
  - still has some unsolved quirks, issues documented in https://github.com/HDR-Development/HewDraw-Remix/issues/1888

### params
- air_brake_y 0.023 -> 0.0225
  -  noticed this value wasn't matching pm's, maybe helps him fall a tiny tad faster? i cant tell, probably negligible
- param_pkthunder.speed: 1.5 -> 1.65
  - gave a slight speed increase to the pkt1 ball projectile to help chose angles better

### CODE REVIEW

**Comment by O1yyy** [comment]
i dont THINK i need `rom_watch.txt` as part of this pr, if anyone knows what's explicitly needed to declare a new `hdr.xml`, please request the changes

otherwise yeah this is ready for review

**Comment by O1yyy** [comment]
i left in an unfinished stall prevention script in ness's `opff.rs`, if anyone with a bigger brain wants to help decipher the logic to make it work i will put it for both ness and lucas

File: fighters/ness/src/acmd/specials.rs [hunk 1]
```diff
@@ -92,16 +124,17 @@ unsafe fn special_lw_hold_game(fighter: &mut L2CAgentBase) {
...
 unsafe fn special_air_lw_hold_game(fighter: &mut L2CAgentBase) {
     let lua_state = fighter.lua_state_agent;
     let boma = fighter.boma();
-    //wait_loop_clear(lua_state);
+   //wait_loop_clear(lua_state);
     for _ in 0..999 {
         if is_excute(fighter) {
-            ATTACK(fighter, 0, 0, Hash40::new("top"), 5.5, 361, 100, 70, 0, 8.5, 0.0, 6.5, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_POS, false, 0, 0.0, 10, false, false, false, false, false, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_MAGIC, *ATTACK_REGION_PSI);
-        }
+            ATTACK(fighter, 0, 0, Hash40::new("top"), 5.5, 361, 35, 0, 70, 2.5, 0.0, 6.5, 2.0, None, None, None, 0.5, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 0, 0.0, 20, false, false, false, false, false, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_MAGIC, *ATTACK_REGION_ENERGY);
```
Comment (WuBoytH) [inline fighters/ness/src/acmd/specials.rs]: Remove the 20 rehit rate on these hitboxes, I know it literally doesn't matter but like it does nothing

File: utils/agent_params.txt [hunk 2]
```diff
@@ -32,4 +32,6 @@ ganon: fighter/ganon/param/hdr.prc
 miifighter: fighter/miifighter/param/hdr.prc
 sheik: fighter/sheik/param/hdr.prc
 miigunner: fighter/miigunner/param/hdr.prc
-falco: fighter/falco/param/hdr.prc
\ No newline at end of file
+miigunner: fighter/miigunner/param/hdr.prc
```
Comment (H3ra-57) [inline utils/agent_params.txt]: remove extra mii gunner

**Review by H3ra-57 — DISMISSED** [review]
make sure `rom_watch.txt` is actually up to date

**Review by hdr-balance-backroom — DISMISSED** [review]
- Document all changes, some are missing such as dthrow's angle and knockback change
- Revert dthrow FAF, readjust knockback to maintain relatively similar combo ability

**Comment by O1yyy** [comment]
> * Document all changes, some are missing such as dthrow's angle and knockback change
> 
>     * Revert dthrow FAF, readjust knockback to maintain relatively similar combo ability

so about this, the dthrow angle was from riah's pr, which is documented there and i mention at the very top, any and all hitbox changes are from that pr, my pr only does what i note down to reduce redundancy

backroom looked at that pr in its completed state, dthrow angle and kb included, and this is all you requested that time:

image

this stuff i already did so thats taken care of 

so, besides reverting dthrow FAF (and the kb which riah already adjusted), all of my changes are documented, and any other solid number changes are back in https://github.com/HDR-Development/HewDraw-Remix/pull/1770 here

**Comment by O1yyy** [comment]
ready for review again, will refrain from touching it so that it can get into a beta

**Comment by Riah321** [comment]
"We'll leave it untouched"

9 commits later....

**Review by WuBoytH — APPROVED** [review]

**Pull request merged**